File: src/test/java/java_/FizzBuzzTest.java
package java_;

import org.junit.*;
import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.MatcherAssert.assertThat;

public class FizzBuzzTest {

	@Test
	public void fizzBuzzJava() {
		interfaces.FizzBuzz sut = new FizzBuzz();
		assertThat(sut.invoke(1), is("1"));
		assertThat(sut.invoke(3), is("Fizz"));
		assertThat(sut.invoke(5), is("Buzz"));
		assertThat(sut.invoke(15), is("FizzBuzz"));
	}

	@Test
	public void fizzBuzzScala() {
		interfaces.FizzBuzz sut = new scala.FizzBuzz();
		assertThat(sut.invoke(1), is("1"));
		assertThat(sut.invoke(3), is("Fizz"));
		assertThat(sut.invoke(5), is("Buzz"));
		assertThat(sut.invoke(15), is("FizzBuzz"));
	}

	@Test
	public void fizzBuzzGroovy() {
		interfaces.FizzBuzz sut = new groovy.FizzBuzz();
		assertThat(sut.invoke(1), is("1"));
		assertThat(sut.invoke(3), is("Fizz"));
		assertThat(sut.invoke(5), is("Buzz"));
		assertThat(sut.invoke(15), is("FizzBuzz"));
	}
}

File: src/test/scala/scala/FizzBuzzTest.scala
package scala

import org.scalatest._
import java_.{FizzBuzz => JFizzBuzz}
import groovy.{FizzBuzz => GFizzBuzz}

class FizzBuzzTest extends FunSpec {

  describe("JavaFizzBuzz") {
    it ("should return 'FizzBuzz' from a multiple of fifty") {
      assert(new JFizzBuzz().invoke(15) == "FizzBuzz")
    }
    it ("should return 'Fizz' from a multiple of three only") {
      assert(new JFizzBuzz().invoke(3) == "Fizz")
    }
    it ("should return 'Buzz' from a multiple of five only") {
      assert(new JFizzBuzz().invoke(5) == "Buzz")
    }
    it ("should return the stringified input from a non multiple of three or five") {
      assert(new JFizzBuzz().invoke(15) == "FizzBuzz")
    }
  }

  describe("ScalaFizzBuzz") {
    it ("should return 'FizzBuzz' from a multiple of fifty") {
      assert(new FizzBuzz().invoke(15) == "FizzBuzz")
    }
    it ("should return 'Fizz' from a multiple of three only") {
      assert(new FizzBuzz().invoke(3) == "Fizz")
    }
    it ("should return 'Buzz' from a multiple of five only") {
      assert(new FizzBuzz().invoke(5) == "Buzz")
    }
    it ("should return the stringified input from a non multiple of three or five") {
      assert(new FizzBuzz().invoke(15) == "FizzBuzz")
    }
  }

  describe("GroovyFizzBuzz") {
    it ("should return 'FizzBuzz' from a multiple of fifty") {
      assert(new GFizzBuzz().invoke(15) == "FizzBuzz")
    }
    it ("should return 'Fizz' from a multiple of three only") {
      assert(new GFizzBuzz().invoke(3) == "Fizz")
    }
    it ("should return 'Buzz' from a multiple of five only") {
      assert(new GFizzBuzz().invoke(5) == "Buzz")
    }
    it ("should return the stringified input from a non multiple of three or five") {
      assert(new GFizzBuzz().invoke(15) == "FizzBuzz")
    }
  }
}

